### PR TITLE
Team referral type was added to the get referrals endpoint

### PR DESCRIPTION
**Pull request by ice-myles**

### CODE REVIEW

File: users/users_referrals.go [hunk 1]
```diff
@@ -49,6 +49,13 @@ func (r *repository) GetReferrals(ctx context.Context, userID string, referralTy
 							AND referrals.id != t1.id
                    			AND referrals.username != referrals.id
 						    AND referrals.referred_by != referrals.id`
+	case TeamReferrals:
+		result, err := r.getTeamReferrals(ctx, userID, limit, offset)
```
Comment (ice-ares) [inline users/users_referrals.go:53]: why couldn't you follow the same pattern and just setup a different join?

File: users/users_referrals.go [hunk 2]
```diff
@@ -160,6 +167,188 @@ func (r *repository) GetReferrals(ctx context.Context, userID string, referralTy
 	}, nil
 }
 
+//nolint:funlen // Long SQL with field list.
+func (r *repository) getTeamReferrals(ctx context.Context, userID string, limit, offset uint64) (*Referrals, error) {
```
Comment (ice-ares) [inline users/users_referrals.go:171]: whats with this mega over engineering? We just need to have a union between both t1 ant t2 referrals

File: users/users_referrals.go (hunk 1 above)
Comment (ice-myles) [reply users/users_referrals.go:53]: Do you mean something like to use:
```
    JOIN USERS referrals
					ON (referrals.referred_by = u.id OR u.referred_by = referrals.id)
                   AND referrals.username != referrals.id
                   AND referrals.referred_by != referrals.id
```
And 
```
JOIN USERS t1
                	ON t1.referred_by = u.ID
					AND t1.id != u.id
                   	AND t1.username != t1.id
                    AND t1.referred_by != t1.id
						JOIN USERS referrals
							ON referrals.referred_by = t1.ID
							AND referrals.id != t1.id
                   			AND referrals.username != referrals.id
						    AND referrals.referred_by != referrals.id
```
together?
Then define referral type by comparing referred_by with another field?

File: users/users_referrals.go (hunk 2 above)
Comment (ice-myles) [reply users/users_referrals.go:171]: Union by SQL, not by 2 different SQL calls for T1 and T2 separately.

Comment (ice-cronus) [reply users/users_referrals.go:171]: Can we move sql generation to dedicated method and then do something like sprintf("select ... union %[1]v union %[2]v", generate(t1), generate(t2)) to keep same sql logic for different requests

**Pull request merged**